src/dashboard/src/components/administration/forms.py

```diff
@@ -519,44 +362,16 @@ def save_config(self):
                     continue
                 delay = str(float(value) * (24 * 60 * 60))
                 config.add_choice(choice_uuid, fprops['chain'], delay_duration=delay, comment=fprops['label'])
-            elif fprops['type'] == 'chain_choice' and fprops.get('find_duplicates', False):
-                # Persist the choice made by the user for each of the existing
-                # chain links with the same name. See #10216 for more details.
-                try:
-                    choice_name = models.MicroServiceChain.objects.get(id=value).description
-                except models.MicroServiceChainLink.DoesNotExist:
-                    pass
-                else:
-                    for i, item in enumerate(get_duplicated_choices(fprops['label'], choice_name)):
-                        comment = '{} (match {} for "{}")'.format(fprops['label'], i + 1, choice_name)
-                        config.add_choice(item[0], item[1], comment=comment)
+            elif fprops['type'] == 'chain_choice' and 'duplicates' in fprops:
+                desc, matches = fprops['duplicates'][value]
+                for i, match in enumerate(matches):
+                    comment = '{} (match {} for "{}")'.format(fprops['label'], i + 1, desc)
+                    config.add_choice(match[0], match[1], comment=comment)
```

Comment:
This is where we generate config items for multiple links with the same name in our XML document, e.g.:

```xml
        <!-- Normalize (match 1 for "Normalize for preservation") -->
        <preconfiguredChoice>
          <appliesTo>cb8e5706-e73f-472f-ad9b-d1236af8095f</appliesTo>
          <goToChain>612e3609-ce9a-4df6-a9a3-63d634d2d934</goToChain>
        </preconfiguredChoice>
        <!-- Normalize (match 2 for "Normalize for preservation") -->
        <preconfiguredChoice>
          <appliesTo>7509e7dc-1e1b-4dce-8d21-e130515fce73</appliesTo>
          <goToChain>612e3609-ce9a-4df6-a9a3-63d634d2d934</goToChain>
        </preconfiguredChoice>
```